### ISSUE TITLE
ConvertEmailToComment not functioning

### ISSUE THREAD

**Issue by elear** [creecros/Mailmagik#47]
Hi,

Email looks like:

```
From: project member <user@email.com>
Subject: [CommentOnTask#60] foo
[...]
content
```
It gets read but not posted as a comment.

But 

```
From: project member <user@email.com>
Subject: [Task#60] foo
[...]
content
```
... gets posted as email to task.

If I stick a throw statement at the beginning Helper/ConvertEmailtoComment.php doAction, it's not invoked.  I presume that this has something to do with a precondition in the object.

My only guess is that somehow project_id is either not being set, or is being cleared.  But it's a guess.  Any ideas?

**Comment by elear** [comment]
Just a little more debugging.  This line in FetchMail->execute() is not getting any matches:

```
        foreach ($this->getProjectIds() as $project_id) {
```

**Comment by alfredbuehler** [comment]
> This line in FetchMail->execute() is not getting any matches:

Then you probably haven't the action _Automatically Convert Emails to Comments_ activated on any project.

**Comment by elear** [comment]
It's enabled (there's only one project).

<img width="928" alt="Screenshot 2024-09-02 at 21 07 19" src="https://github.com/user-attachments/assets/7acb5bba-b3de-4212-b31e-e99110b4de4d">

**Comment by elear** [comment]
If you have any recommended debugging steps, I'd be grateful.

**Comment by creecros** [comment]
Are you running both actions? Simultaneously.

**Comment by creecros** [comment]
> If you have any recommended debugging steps, I'd be grateful.

I use error_log()

https://www.php.net/manual/en/function.error-log.php

Just plug it in to the code where you want to know what's going on.

**Comment by creecros** [comment]
Rereading your initial post.

You make 2 claims.
1. `Subject: [CommentOnTask#60] foo` does not work
2. `Subject: [Task#60] foo` works

My observation:
It's hard for me to believe that #60 is both the project id and task id. 

The first action requires a task id.
The second actuin requires a project id.

Why are they both #60?

**Comment by elear** [comment]
Thanks for your comments.  60 is the task id, and it really does get parsed properly for [Task#], because I see that it gets properly trimmed in the notification.  It does NOT get properly trimmed for [CommentOnTask#]. 

I'll have a poke with error_log() as you suggest.  I'm only trying one thing at a time, fyi.

**Comment by creecros** [comment]
Task# is not relatable to an action. 
CommentOnTask# is.

I would start with confirming your cron is actually running.

**Comment by creecros** [comment]
> It's enabled (there's only one project).
> 
> <img alt="Screenshot 2024-09-02 at 21 07 19" width="928" src="https://private-user-images.githubusercontent.com/1016623/363777054-7acb5bba-b3de-4212-b31e-e99110b4de4d.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjUzMDc4MzAsIm5iZiI6MTcyNTMwNzUzMCwicGF0aCI6Ii8xMDE2NjIzLzM2Mzc3NzA1NC03YWNiNWJiYS1iM2RlLTQyMTItYjMxZS1lOTkxMTBiNGRlNGQucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDkwMiUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDA5MDJUMjAwNTMwWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9NzZhOTRiMWQ5OTdiNDc5OTY5ZTIzZTBmODJkMWE4MTg4NTJlNGM4Yzk0MDZmZWEwYTQyMWVlZDI4ZjI3M2JjMCZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.1yfmPzyRmB9Ao997IdC2IgioenAK1iiWCJNNf-h1n9o">

And here I notice you are using the cron.

How often is that set to run? Have you tested it, and you know it's running?

**Comment by elear** [comment]
Yep.  cron is running.

**Comment by creecros** [comment]
K. I'll test next chance I get. See what's going  on.

**Comment by alfredbuehler** [comment]
> Yep. cron is running.

How confirmed?

BTW, which versions of Mailmagik are you using?

**Comment by alfredbuehler** [comment]
Now I see the issue. I'll come back soon with a solution.

**Comment by alfredbuehler** [comment]
OK, I have fixed this in my working copy for the upcoming release. 
The problem is a collision in subject parsing, both are matching:  CommentOn<b>Task#</b>60

If Taskmail (Task#) obviously is parsed first, it marks the mail as seen, or deletes it, depending on your settings. Then there is no unseen mail for CommentOnTask#.

I've solved this with the following fix in the file Helper/MailHelper.php:

![image](https://github.com/user-attachments/assets/d19de837-18ef-481c-9c58-e87a0ff92a2b)

Of course, it will be contained in the upcoming release, but you can apply this one-liner for now.

Sorry for the circumstances and thanks for reporting.

**Comment by alfredbuehler** [comment]
I just saw that this is fixed in 1.5.0, are you using an older version? I'm confused, sorry.

**Comment by elear** [comment]
1.5.0 according to the changelog.  What I have been doing for test is to invoke ./cli Mailmagik:fetchmail manually when I send off notes, so that it kicks off the IMAP fetch.  I've left the other cron entry as per normal.  I don't know Symfony well enough to get around your NullOutput trick, but I'll try the log function that was suggested in a bit.  I also don't have a good handle on the entire workflow.  I have some of it, though.  Still working on it.

**Comment by alfredbuehler** [comment]
As I said, I'm confused. It's NOT fixed in my checkout 1.5.0. Please apply the change I mentioned above.

**Comment by elear** [comment]
ah.  Ok

**Comment by elear** [comment]
@alfredbuehler That did the trick.  THANK YOU.

**Comment by alfredbuehler** [comment]
Fixed by 1.5.2
